event/src/CocktailGenerator.cxx

```diff
@@ -109,6 +114,14 @@ CocktailGenerator::Event_t *CocktailGenerator::NextEvent()
     // geometry state not yet initialized
     // Add track to the current event
     event->AddPrimary(track);
+
+    // Count particles
+    observedParticles[particle->GetName()]++;
   }
+
+  // Print particles
```

Comment:
We can complete remove printing in the generator itself since now we can print in the unit test

Reply:
Fixed. Changed generator back to original implementation, unit test now prints the different particles count.